content/swift/concepts/swift_closures.md

```diff
@@ -0,0 +1,203 @@
+---
+Title: 'Closures'
+Description: 'Closures allows for more flexible code.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Closures'
+Catalogue Content:
+  - 'learn-swift'
+  - 'paths/build-ios-apps-with-swiftui'
+---
+
+# Swift Closures
+
+Similar to blocks in C or lambdas in other programming languages, closures in Swift contain chunks of code that can be passed around and reused within the same code. 
+
+They can store and capture references to variables and constants which is also referred to as closing over constants and variables. The best part is that all memory management is handled by Swift. 
+
+## Using Closures
+
+Closures look very similar to functions. To show this, a variable called `hello` is assigned to a closure that prints out "Hello World":
+
+```swift
+let hello = { print("Hello World!") }
+
+hello()
+```
+The closure is the code between the curly brackets, `{` and `}`. The closure is then called by using the opening and closing parenthesis, `(` and `)`, on the constant `hello`. This executes the closure and prints out `Hello World!`.
+
+### Parameters
+
+Just like functions, closures can also take in parameters or input values. The previous example can be changed to say "Hello" to a specific person instead of the generic "Hello World".
+
+```swift
+let helloName:(String) -> () = { name in 
+    print("Hello, \(name)!")
+}
+
+helloName("Sonny")
+```
+This prints out:
+
+`Hello, Sonny!` 
+
+There are three important parts to this closure:
+- The type, `(String) -> ()`, which specifies the value being taken in and returned by the closure 
+- The expression, `{name in ...}`, where `in` separates the parameters from the closure body
+- The call, `hello(...)`, executes the closure
+
+It is important to note the return type is void because nothing of a specific type is being returned in this example. This can be written as `()` or `(void)`
```

Comment:
```suggestion
It is important to note the return type is void because nothing of a specific type is being returned in this example. This can be written as `()` or `(void)`.
```